File: api/app/Models/Province.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Province extends Model
{
    protected $fillable = ['name', 'code', 'is_active'];

    protected $casts = [
        'is_active' => 'boolean',
    ];

    public function districts(): HasMany
    {
        return $this->hasMany(District::class, 'province_id');
    }

    public function customers(): HasMany
    {
        return $this->hasMany(Customer::class, 'province_id');
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class, 'province_id');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Catalog/Entities/Province.cs
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Catalog.Entities;

public sealed class Province : AuditableEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;

    public ICollection<District> Districts { get; set; } = new List<District>();
    public ICollection<Customer> Customers { get; set; } = new List<Customer>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();
    public ICollection<Warehouse> Warehouses { get; set; } = new List<Warehouse>();
}
